### PR TITLE
Cache folder

### PR DESCRIPTION
**Pull request by hadim**
Here is a PR.

The code is here for the cache folder but I haven't been able to test it since I don't know how to execute a main class in Eclipse using Maven so the Maven properties set cannot be read... Usually, I just use the goal "Java Application" and it works just well.

Also, I have added a commit that allows loading any file from within the archive.

### CODE REVIEW

**Comment by hadim** [comment]
Also, this is absolutely not urgent @ctrueden. So take your time on this one.

File: src/main/java/net/imagej/tensorflow/DefaultTensorFlowService.java [hunk 1]
```diff
@@ -66,15 +66,17 @@
 import org.tensorflow.Graph;
 import org.tensorflow.SavedModelBundle;
 
+import scala.tools.jline_embedded.internal.Log;
```
Comment (frauzufall) [inline src/main/java/net/imagej/tensorflow/DefaultTensorFlowService.java:69]: This import should be removed

File: src/main/java/net/imagej/tensorflow/DefaultTensorFlowService.java [hunk 2]
```diff
@@ -191,17 +198,27 @@ public void dispose() {
...
 	private synchronized void initModelCache() {
 		final DiskLocationCache cache = new DiskLocationCache();
 
 		// Cache the models into $IMAGEJ_DIR/models.
-		final File baseDir = appService.getApp().getBaseDirectory();
-		final File cacheBase = new File(baseDir, "models");
-		if (!cacheBase.exists()) cacheBase.mkdirs();
+		final File cacheBase;
+
+		System.out.println(System.getProperty(CACHE_DIR_PROPERTY_KEY));
```
Comment (frauzufall) [inline src/main/java/net/imagej/tensorflow/DefaultTensorFlowService.java:212]: Maybe we either use the log service or just remove this line as well.

**Review by frauzufall — COMMENTED** [review]
Hey, I reviewed the PR because I would love to use the `loadFile()`method that comes with it. We have another settings file bundled with the model and currently I have no way to get the model cache location.. Anyways, I made two small comments, but I was also not successful in testing it because I cannot manage to set the system variable.. Here is my `.m2/settings.xml` file:

```
<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0 http://maven.apache.org/xsd/settings-1.0.0.xsd">
  <pluginGroups></pluginGroups>
  <proxies></proxies>
  <servers></servers>
  <mirrors></mirrors>
  <profiles>
	<profile>
		<id>random-dev</id>
		<properties>
			<imagej.models.dir>/home/random/Development/mymodels/</imagej.models.dir>
		</properties>
	</profile>
  </profiles>

  <activeProfiles>
    <activeProfile>random-dev</activeProfile>
  </activeProfiles>
</settings>
```
But `System.getProperty("imagej.models.dir")` returns null at [this line](https://github.com/hadim/imagej-tensorflow/blob/a9107d8658596ab26db2934d200fd02eaabf4ebb/src/main/java/net/imagej/tensorflow/DefaultTensorFlowService.java#L212). This is the first time I am using the maven settings file so maybe I misunderstood something.

**Comment by frauzufall** [comment]
I was able to add `-Dimagej.models.dir=/home/random/Development/mymodels` as VM options in IntelliJ (@hadim you can do that in Eclipse too) and then it worked as expected.

Same way to change it by starting ImageJ via command line:
```
./ImageJ-linux64 -Dimagej.models.dir=/home/random/Development/mymodels
```

**Comment by hadim** [comment]
The most important function for me was `loadFile()` and so I ended up using a custom `DownloadService` that does the job: https://github.com/hadim/maskflow-fiji/blob/bd2a21c01813363056e2ddd42e4325d169bbd987/src/main/java/sc/fiji/maskflow/CustomDownloadService.java

I would suggest you do the same. With this you can just do:

```java
@Parameter
private CustomDownloadService cds;
// ...
File parametersFile = cds.loadFile(this.modelLocation, this.modelnameCache, "config.yml");
```

**Comment by hadim** [comment]
@frauzufall you force pushed on `master`! You should never do that on a `master` branch (it's ok on PR branches). Even if it's to remove the last commit. Instead, you should push a new commit that reverts the last commit.

Since things have already been updated on Maven I would suggest you bump the version to `1.0.2-SNAPSHOT`. I can't do it myself, cuz I don't have write access on this repo.

**Comment by hadim** [comment]
(looks like you're learning the hard way :-D)

**Comment by frauzufall** [comment]
:sob: I uploaded a new version. So sorry! Will not happen again. Would you create a new PR?

**Comment by ctrueden** [comment]
@hadim I added you to this repo's team.

@frauzufall Thank you for moving this PR forward!

I would rather avoid custom services whenever possible. Would be nice to improve the standard `DownloadService` etc. to support needed use cases.

**Pull request merged**